File: 24/cpp/main.cpp
#include <iostream>

class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        // Edge case: If the list is empty or has only one element, return as is
        if (!head || !head->next) return head;

        // Initialize a dummy node that helps track the new head
        ListNode* dummy = new ListNode(0);
        dummy->next = head;
        ListNode* prev = dummy;

        while (head && head->next) {
            // Nodes to be swapped
            ListNode* first = head;
            ListNode* second = head->next;

            // Swapping the pairs
            first->next = second->next;
            second->next = first;
            prev->next = second;

            // Move to the next pair
            prev = first;
            head = first->next;
        }

        return dummy->next;
    }
};

int main() {
    Solution sol;

    // Test case X: Y
    RETURN_TYPE resultX = sol.FUNCTION_NAME(Y);
    std::cout << "Test X: " << resultX << std::endl;

    return 0;
}

File: 24/python3/main.py
#!/usr/bin/env python3

class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        # Edge case: if the list is empty or has only one node
        if not head or not head.next:
            return head

        # Dummy node to simplify edge case handling
        dummy = ListNode(0)
        dummy.next = head
        prev = dummy

        while head and head.next:
            # Nodes to be swapped
            first = head
            second = head.next

            # Swapping the pairs
            first.next = second.next
            second.next = first
            prev.next = second

            # Move to the next pair
            prev = first
            head = first.next

        return dummy.next
